File: Insert an element 3 .c
#include <stdio.h>
#include <string.h>
#include <math.h>
#include <stdlib.h>

int main() {

    int n;
    scanf("%d",&n);
    int a[n+1],i;
    for(i=0;i<n;i++)
    {
        scanf("%d",&a[i]);
    }
    int pos,val;
    scanf("%d%d",&pos,&val);
    if(pos>n){
        printf("Invalid Input");
}
    else
{
        for(i=n;i>=pos;i--)
        {
            a[i]=a[i-1];
        }
        a[pos-1]=val;
        for(i=0;i<=n;i++)
        {
            printf("%d\n",a[i]);
        }
}
    return 0;
}

File: Insert an element3.java
import java.io.*;
import java.util.*;

public class Solution {

    public static void main(String[] args) {
         Scanner s=new Scanner(System.in);
        int n;
        n=s.nextInt();
        int a[]=new int[n+1];
        for(int i=0;i<n;i++)
        {
            a[i]=s.nextInt();
        }
        int pos,val;
        pos=s.nextInt();
        val=s.nextInt();
        if(pos>n)
        {
            System.out.println("Invalid Input");
        }
        else{
            for(int i=n;i>=pos;i--)
            {
                a[i]=a[i-1];
            }
            a[pos-1]=val;
            for(int i=0;i<=n;i++)
            {
                System.out.println(a[i]);
            }
        }
    }
}


